### DISCUSSION TITLE
why lost-pixel is looking for an old version of playwright/chrome?

### DISCUSSION THREAD

**Discussion by studioab** [lost-pixel/lost-pixel#298] [General]
this pipeline in Azure has been working without any issues until last week.

previous task is installing latest chrome
<img width="485" alt="image" src="https://github.com/lost-pixel/lost-pixel/assets/13108111/0e46982c-e6d6-4b3b-a499-20f1141717cc">

next task with lost-pixel fails
![image](https://github.com/lost-pixel/lost-pixel/assets/13108111/d656221b-31b8-4f9a-81a2-54da4604c3a1)

**Comment by studioab** [comment]
downgrading works but it seems wrong 
npx playwright@1.37.1 install

**Comment by d-ivashchuk** [comment]
are you on the latest lost-pixel version? namely 3.6.0 as of today

**Comment by studioab** [comment]
![image](https://github.com/lost-pixel/lost-pixel/assets/13108111/1323e0f0-6d13-42b0-b0c4-f4d478ded00e)

**Comment by d-ivashchuk** [comment]
are you running it locally? usually after we update the version of the playwright it needs to be updated as well. we will try to see what we can automate here

**Reply by studioab** [reply]
it is failing only in the azure build pipeline

**Comment by Mattsi-Jansky** [comment]
Same issue here, unfortunately. Thought I'd throw it at my app that already has Storybook and see what happened. But Playwright installs `chromium-1117`, which results in Lost Pixel erroring because it is unable to find an older version, `chromium-1091`.

Output from `npx playwright install`:

```
Removing unused browser at /home/mattsi/.cache/ms-playwright/chromium-1105
Removing unused browser at /home/mattsi/.cache/ms-playwright/firefox-1440
Removing unused browser at /home/mattsi/.cache/ms-playwright/webkit-1983
Downloading Chromium 125.0.6422.26 (playwright build v1117) from https://playwright.azureedge.net/builds/chromium/1117/chromium-linux.zip
156.8 MiB [====================] 100% 0.0s
Chromium 125.0.6422.26 (playwright build v1117) downloaded to /home/mattsi/.cache/ms-playwright/chromium-1117
Downloading Firefox 125.0.1 (playwright build v1449) from https://playwright.azureedge.net/builds/firefox/1449/firefox-ubuntu-22.04.zip
85.7 MiB [====================] 100% 0.0s
Firefox 125.0.1 (playwright build v1449) downloaded to /home/mattsi/.cache/ms-playwright/firefox-1449
Downloading Webkit 17.4 (playwright build v2003) from https://playwright.azureedge.net/builds/webkit/2003/webkit-ubuntu-22.04.zip
84.8 MiB [====================] 100% 0.0s
Webkit 17.4 (playwright build v2003) downloaded to /home/mattsi/.cache/ms-playwright/webkit-2003
```

Subsequent output from `npx lost-pixel update`:

```
Version: 3.16.0
Loading project config ...
Current working directory: <redacted>
Looking for config file: <redacted>/lostpixel.config.(ts|js|cjs|mjs)
✅ Found config file: <redacted>/lostpixel.config.ts
🚀 Starting Lost Pixel in 'generateOnly' mode
Running lost-pixel in update mode. Baseline screenshots will be updated
📂 Creating shot folders
📸 Creating shots
Removing 0 files from .lostpixel/current/
Removing 0 files from .lostpixel/difference/

=== [Storybook Mode] ./storybook-static ===

❌ browserType.launch: Executable doesn't exist at /home/mattsi/.cache/ms-playwright/chromium-1091/chrome-linux/chrome
╔═════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright Test or Playwright was just installed or updated. ║
║ Please run the following command to download new browsers:              ║
║                                                                         ║
║     npx playwright install                                              ║
║                                                                         ║
║ <3 Playwright Team                                                      ║
╚═════════════════════════════════════════════════════════════════════════╝
Sending anonymized telemetry data.
```

Note that Playwright has installed Chrome in `~/.cache/ms-playwright/chromium-1117` but Lost Pixel is looking for it in `~/.cache/ms-playwright/chromium-1091/`.

Using `"@playwright/test": "1.44.1",` installed explicitly as a dev dependency because I use it for other things too, and `"lost-pixel": "3.16.0",`. Ubuntu 22.04.4 LTS.

**Comment by baoengineer** [comment]
Facing the same issue, on CI I also need to use docker image: mcr.microsoft.com/playwright:v1.40.0-focal this specific version to have that chromium-1091 version to work.

**Comment by silvioprog** [comment]
Faced the same problem on Fedora 41 and fixed it by specifying the `executablePath` explicitly:

```typescript
// lostpixel.config.ts

import type { CustomProjectConfig } from 'lost-pixel';
import { chromium } from 'playwright-core';

export const config: CustomProjectConfig = {
  browserLaunchOptions: {
    chromium: {
      executablePath: chromium.executablePath(),
      headless: false,
      args: ['--headless']
    }
  },
  storybookShots: {
    storybookUrl: 'http://127.0.0.1:6006',
  },
  generateOnly: true,
  failOnDifference: process.env.LOST_PIXEL_IS_UPDATING !== '1'
};
```

```raw
lost-pixel: ^3.22.0
```

It is using Chrome from `~/.cache/ms-playwright/chromium-1155/chrome-linux/chrome`.

**Reply by AndyClausen** [reply]
Absolute life saver, cheers. Worked great for me after adding `playwright-core` as an explicit dependency.
